fooof/objs/fit.py

```diff
@@ -431,6 +441,14 @@ def fit(self, freqs=None, power_spectrum=None, freq_range=None):
         # In rare cases, the model fails to fit, and so uses try / except
         try:
 
+            # If not set to fail on NaN or Inf data at add time, check data here
+            #   This serves as a catch all for curve_fits which will fail given NaN or Inf
+            #   Because FitError's are by default caught, this allows fitting to continue
+            if not self._check_data:
+                if np.any(np.isinf(self.power_spectrum)) or np.any(np.isnan(self.power_spectrum)):
+                    raise FitError("There are NaN or Inf values in the data,  "
```

Comment:
I think debug mode can be removed. It is more of a DataError since it is a data check prior to fitting. This check is identical to DataError in `_check_data` raised in 1215. One would also need to set check mode to True and data mode to False, for debug to work:
```
fm = FOOOF()
fm.set_check_data_mode(False)
fm.set_debug_mode(True)
fm.fit(freqs, pows)
```
And after this, the error raised is less verbose than raised by the DataError in line 1216.

Reply:
I'm not sure I'm following what you're suggesting. 

The goal of `debug` mode is to see trigger an error if a model is not fit, to see why it's failing. 
For example:
```
tfm = FOOOF(verbose=False)
tfm._maxfev = 5  # This mimics a particular kind of fit failure
tfm.set_debug_mode(True)
tfm.fit(freqs, pows)
tfm.get_results()
```

When debug mode is off, the results are NaN. If one wants to see why this model is not fitting, turn debug mode on, and the error will be raised to see where it is. 

I think debug mode is useful (though a similar outcome could arguable be achieved by just logging errors). 

The interaction between debug mode and check_data mode is slightly tricky. If check_data is off, there is no data checking (no DataError). I think turning on debug mode, or turning off data check mode are both useful, though it's not super clear if one would want to combination (because in that case, one already knows why most null models are null). But, for consistency, for tracking FitErrors, this error here indicates why a particular fit failed. This particular only triggers in the case that check_data is False and debug is True.

Reply:
Sorry for the confusing comment. I see the usefulness of debug mode now. My comment meant to ask about the three different errors/outputs that are possible for nan's in data, depending on settings:

```
DataError: The input power spectra data, after logging, contains NaNs or Infs. This will cause the fitting to fail. One reason this can happen is if inputs are already logged. Inputs data should be in linear spacing, not log.
```

```
FitError: There are NaN or Inf values in the data,  which preclude model fitting.
```

```
Model fitting was unsuccessful.
```

These make sense after going back through the code and your comment.